app/workers/deletes.php

```diff
@@ -544,27 +544,28 @@ protected function deleteBucket(Document $document, string $projectId)
         $dbForProject = $this->getProjectDB($projectId);
         $dbForProject->deleteCollection('bucket_' . $document->getInternalId());
 
-        $device = new Local(APP_STORAGE_UPLOADS.'/app-'.$projectId);
-        
-        switch (App::getEnv('_APP_STORAGE_DEVICE', Storage::DEVICE_LOCAL)) {
+        $device = App::getEnv('_APP_STORAGE_DEVICE', Storage::DEVICE_LOCAL);
+        $root = APP_STORAGE_UPLOADS . '/app-' . $projectId;
+        switch ($device) {
             case Storage::DEVICE_S3:
-                $s3AccessKey = App::getEnv('_APP_STORAGE_S3_ACCESS_KEY', '');
-                $s3SecretKey = App::getEnv('_APP_STORAGE_S3_SECRET', '');
-                $s3Region = App::getEnv('_APP_STORAGE_S3_REGION', '');
-                $s3Bucket = App::getEnv('_APP_STORAGE_S3_BUCKET', '');
-                $s3Acl = 'private';
-                $device = new S3(APP_STORAGE_UPLOADS . '/app-' . $projectId, $s3AccessKey, $s3SecretKey, $s3Bucket, $s3Region, $s3Acl);
-                break;
             case Storage::DEVICE_DO_SPACES:
-                $doSpacesAccessKey = App::getEnv('_APP_STORAGE_DO_SPACES_ACCESS_KEY', '');
-                $doSpacesSecretKey = App::getEnv('_APP_STORAGE_DO_SPACES_SECRET', '');
-                $doSpacesRegion = App::getEnv('_APP_STORAGE_DO_SPACES_REGION', '');
-                $doSpacesBucket = App::getEnv('_APP_STORAGE_DO_SPACES_BUCKET', '');
-                $doSpacesAcl = 'private';
-                $device = new DOSpaces(APP_STORAGE_UPLOADS . '/app-' . $projectId, $doSpacesAccessKey, $doSpacesSecretKey, $doSpacesBucket, $doSpacesRegion, $doSpacesAcl);
+            case Storage::DEVICE_BACKBLAZE:
+            case Storage::DEVICE_LINODE:
+            case Storage::DEVICE_WASABI:
+                $accessKey = App::getEnv('_APP_STORAGE_S3_ACCESS_KEY', '');
+                $secretKey = App::getEnv('_APP_STORAGE_S3_SECRET', '');
+                $region = App::getEnv('_APP_STORAGE_S3_REGION', '');
+                $bucket = App::getEnv('_APP_STORAGE_S3_BUCKET', '');
+                $acl = 'private';
+                /**@var $adapter Utopia\Storage\Device**/
+                $adapter = 'Utopia\\Storage\\Device\\' . $device;
```

Comment:
```suggestion
                $adapter = 'Utopia\\Storage\\Device\\' . \ucfirst($device);
```

We should have proper case everywhere

Reply:
Cool, but in case of  digital ocean (DOSpaces), back blaze(BackBlaze) there is no protection from user typeo.

Reply:
The current  flow is that when there is no match in the device name it falls to Local.
To change it to adapter class exist validation?

Reply:
Well let's keep it as is then, and let's other's review. Make sure to resolve conflicts so that we can see the tests running.

Reply:
ok